paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/AutoTagForSavepointCommitterOperator.java

```diff
@@ -127,8 +127,10 @@ public OperatorSnapshotFutures snapshotState(
     @Override
     public void notifyCheckpointComplete(long checkpointId) throws Exception {
         commitOperator.notifyCheckpointComplete(checkpointId);
-        if (identifiersForTags.remove(checkpointId)) {
-            createTagForIdentifiers(Collections.singletonList(checkpointId));
+        // savepoint do not trigger notifyCheckpointComplete method
+        long savepointId = checkpointId - 1;
```

Comment:
We may need to make tags for all previous successful savepoints that are less than `checkpointId`, instead of just `checkpointId - 1 `(considering the case where multiple savepoints are triggered consecutively).  
This feature may work fine under `stop-with-savepoint`.